core/src/main/java/org/elasticsearch/search/aggregations/bucket/nested/NestedAggregator.java

```diff
@@ -139,13 +146,28 @@ private static Query findClosestNestedPath(Aggregator parent) {
 
     public static class Factory extends AggregatorFactory {
 
-        private final String path;
+        private String path;
 
-        public Factory(String name, String path) {
+        public Factory(String name) {
             super(name, InternalNested.TYPE);
+        }
+
+        /**
+         * Set the path to use for this nested aggregation. The path must match
+         * the path to a nested object in the mappings. This parameter is
+         * compulsory.
```

Comment:
I'm curious why you moved `path` aways from the constructor if it is compulsory?